### PR TITLE
[FLINK-35559][Connector/Kinesis] fix shading configuration to avoid class conflicts when using flink-connector-kinesis alongside Firehose sink

### PR DESCRIPTION
**Pull request by z3d1k**
## Purpose of the change

Add `org.apache.flink.connector.aws.sink` to relocation list in order to prevent class conflicts while including `flink-connector-kinesis` alongside with `flink-connector-aws-kinesis-firehose`.

## Verifying this change

- Manually verified by running the Kinesis connector and Firehose sink on a local Flink cluster.
- Verified build artifacts

## Significant changes
*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterwards, for convenience.)*
- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [ ] New feature has been introduced
  - If yes, how is this documented? not applicable

### CODE REVIEW

File: flink-connector-aws/flink-connector-kinesis/pom.xml [hunk 1]
```diff
@@ -354,6 +354,12 @@ under the License.
                             <relocations combine.children="override">
                                 <!-- Do not relocate guava because it is exposed in the Kinesis API (KinesisProducer#addUserRecord).
                                      Users may be using other affected API's, so relocations may break user-code -->
+                                <relocation>
+                                    <pattern>org.apache.flink.connector.aws.sink</pattern>
+                                    <shadedPattern>
+                                        org.apache.flink.kinesis.shaded.org.apache.flink.connector.aws.sink
+                                    </shadedPattern>
```
[the remark this answers is not in the document]
Comment (foxus) [reply flink-connector-aws/flink-connector-kinesis/pom.xml:361]: @hlteoh37 are you referring to the risk of restoring state that references FQCN that has now changed?

[the remark this answers is not in the document]
Comment (z3d1k) [reply flink-connector-aws/flink-connector-kinesis/pom.xml:361]: @hlteoh37 No, this package currently contains only util classes for working with AWS exceptions and not used for state.

[the remark this answers is not in the document]
Comment (hlteoh37) [reply flink-connector-aws/flink-connector-kinesis/pom.xml:361]: Ok sounds good - we discussed offline and suggested changing the shading in `flink-connector-kinesis` to include the whole `flink-connector-aws-base` module instead.

[the remark this answers is not in the document]
Comment (z3d1k) [reply flink-connector-aws/flink-connector-kinesis/pom.xml]: updated

**Review by dannycranmer — APPROVED** [review]

**Comment by dannycranmer** [comment]
There is a risk this will break user code depending on these libs, right? Given this is a bug, and they should not be using the shaded libs, I think this is ok, thoughts?

**Review by hlteoh37 — APPROVED** [review]

**Pull request merged**